pkg/config/mesh/networks_watcher.go

```diff
@@ -32,18 +32,28 @@ type NetworksHolder interface {
 	PrevNetworks() *meshconfig.MeshNetworks
 }
 
+// WatcherHandler is a wrapper for Handler func, in order to
+// unregister handler when remote cluster is removed.
+type WatcherHandler struct {
+	Handler func()
+}
+
 // NetworksWatcher watches changes to the mesh networks config.
 type NetworksWatcher interface {
 	NetworksHolder
 
-	AddNetworksHandler(func())
+	// AddNetworksHandler registers a callback handler for changes to the networks config.
+	AddNetworksHandler(*WatcherHandler)
```

Comment:
nit: informer is like `AddEventHandler(handler ResourceEventHandler) (ResourceEventHandlerRegistration, error)`. Maybe we should use the same (return a handle) instead of mutating? Should behave the same in practice, though

Reply:
Good suggestion.